### ISSUE TITLE
*** Error in `python': malloc(): memory corruption:

### ISSUE THREAD

**Issue by Leon924** [dmlc/gluon-cv#758]
(tvm36) [liqiang@inspur ssd]$ python train_ssd.py 
[11:00:12] src/operator/nn/mkldnn/mkldnn_base.cc:74: Allocate 147456 bytes with malloc directly
[11:00:12] src/operator/nn/mkldnn/mkldnn_base.cc:74: Allocate 589824 bytes with malloc directly
[11:00:12] src/operator/nn/mkldnn/mkldnn_base.cc:74: Allocate 2359296 bytes with malloc directly
[11:00:12] src/operator/nn/mkldnn/mkldnn_base.cc:74: Allocate 9437184 bytes with malloc directly
INFO:root:Namespace(batch_size=16, data_shape=512, dataset='voc', epochs=240, gpus='1,2', log_interval=100, lr=0.001, lr_decay=0.1, lr_decay_epoch='160,200', momentum=0.9, network='resnet50_v1', num_workers=4, resume='', save_interval=10, save_prefix='ssd_512_resnet50_v1_voc', seed=233, start_epoch=0, syncbn=False, val_interval=1, wd=0.0005)
INFO:root:Start training from [Epoch 0]
python: malloc.c:2365: sysmalloc: Assertion `(old_top == (((mbinptr) (((char *) &((av)->bins[((1) - 1) * 2])) - __builtin_offsetof (struct malloc_chunk, fd)))) && old_size == 0) || ((unsigned long) (old_size) >= (unsigned long)((((__builtin_offsetof (struct malloc_chunk, fd_nextsize))+((2 * (sizeof(size_t))) - 1)) & ~((2 * (sizeof(size_t))) - 1))) && ((old_top)->size & 0x1) && ((unsigned long)old_end & pagemask) == 0)' failed.
*** Error in `python': malloc(): memory corruption: 0x00007ff4a006df30 ***
python: malloc.c:3609: _int_malloc: Assertion `(unsigned long)(size) >= (unsigned long)(nb)' failed.
*** Error in `python': malloc(): memory corruption: 0x00007ff4a0747320 ***

what is this? anybody help?

**Comment by lsabi** [comment]
Are you using an intel cpu with the mxnet intel library? Because I used to get the same error and the only solution was to switch to the normal version.

**Comment by zhreshold** [comment]
If you are using mxnet-mkl version and get this error, try upgrade or use mxnet without mkl

**Comment by petit-ami** [comment]
@lsabi @zhreshold  after change it  into mxnet-cu90, It works fine. 
Thank you!

**Comment by pengzhao-intel** [comment]
@xinyu-intel, please take a look for this issue.

@petit-ami thanks to reporting the problem, we will fix it :)

**Comment by xinyu-intel** [comment]
@petit-ami cannot reproduce this bug by using `pip install mxnet-cu90mkl`. Did you build mxnet from source? It seems mxnet was built with `tcmalloc`. You can try to build with `USE_GPERFTOOLS=0` or just git pull the lastest mxnet which has already set `tcmalloc` to false by default.

**Comment by lsabi** [comment]
@xinyu-intel when it occurred to me, I was using mxnet-mkl (no, gpu ...nvidia driver support on kubuntu is really bad). And I was also finetuning the SSD nn, with more than 2 epochs. I remember it was strange, because, for example, with `for epoch in range(0, 2):` it worked, but with `for epoch in range(0, 3):` it  did not work.

If I still have the file I'll upload it, though it's almost the same (I changed the dataloader) from the example [fine-tune gluon-cv](https://gluon-cv.mxnet.io/build/examples_detection/finetune_detection.html#sphx-glr-build-examples-detection-finetune-detection-py)
